Mu2eG4/src/constructDiskCalorimeter.cc

```diff
@@ -289,48 +288,117 @@ namespace mu2e {
+           G4RotationMatrix* sPipeRotate1 = new G4RotationMatrix();
+           sPipeRotate1 -> rotateX(0.5 * CLHEP::pi);
+           sPipeRotate1 -> rotateY((-0.5*CLHEP::pi + zRotateAngle) * CLHEP::rad);
+           G4RotationMatrix* sPipeRotate2 = new G4RotationMatrix();
+           sPipeRotate2 -> rotateX(0.5 * CLHEP::pi);
+           sPipeRotate2 -> rotateY((-0.5*CLHEP::pi - zRotateAngle) * CLHEP::rad);
+           G4RotationMatrix* sPipeRotate3 = new G4RotationMatrix();
+           sPipeRotate3 -> rotateX(0.5 * CLHEP::pi);
+           sPipeRotate3 -> rotateY((-1.5*CLHEP::pi + zRotateAngle) * CLHEP::rad);
+           G4RotationMatrix* sPipeRotate4 = new G4RotationMatrix();
+           sPipeRotate4 -> rotateX(0.5 * CLHEP::pi);
+           sPipeRotate4 -> rotateY((-1.5*CLHEP::pi - zRotateAngle) * CLHEP::rad);
+           //reg.add(sPipeRotate);
+           // transform movement
+           double xCenter = 0.5 * (xEnd + xStart);
+           double yCenter = 0.5 * (yEnd + yStart);
+           // the strait tubes at the origin coordinate system
+           //std::cout << "the length of the strait pipes: " << sLength << std::endl; 
+           G4Tubs* pipe3 = new G4Tubs("caloPipe3", pipeRadius-pipeThickness, pipeRadius, 0.5 * (sLength - 4.), 0, CLHEP::twopi);
+           G4LogicalVolume* pipe3Log = caloBuildLogical(pipe3, pipeMaterial, "caloPipe3Log",isPipeVisible,G4Color::Cyan(),isPipeSolid,forceEdge);
+   
+           // large bending torus
+           pv = new G4PVPlacement(rotPipe1, G4ThreeVector(0,0,z), pipe1Log, "caloPipePV", frontPanelFoamLog, false, ipipe, false);
            doSurfaceCheck && checkForOverlaps( pv, config, verbosityLevel>0);                
-           pv = new G4PVPlacement(rotPipe2,G4ThreeVector(0,0,z), pipe1Log, "caloPipePV", frontPanelFoamLog, false, ipipe, false);
-           doSurfaceCheck && checkForOverlaps( pv, config, verbosityLevel>0);                                
-           pv = new G4PVPlacement(rotPipeFlat,G4ThreeVector(xpipe, y0,z), pipe2Log, "caloPipePV", frontPanelFoamLog, false, ipipe , false);
-           doSurfaceCheck && checkForOverlaps( pv, config, verbosityLevel>0);           
-           pv = new G4PVPlacement(rotPipeFlat,G4ThreeVector(xpipe,-y0,z), pipe2Log, "caloPipePV", frontPanelFoamLog, false, ipipe , false);
-           doSurfaceCheck && checkForOverlaps( pv, config, verbosityLevel>0);           
-           pv = new G4PVPlacement(rotPipeFlat,G4ThreeVector(-xpipe,y1,z), pipe2Log, "caloPipePV", frontPanelFoamLog, false, ipipe , false);
-           doSurfaceCheck && checkForOverlaps( pv, config, verbosityLevel>0);           
-           pv = new G4PVPlacement(rotPipeFlat,G4ThreeVector(-xpipe,-y1,z), pipe2Log, "caloPipePV", frontPanelFoamLog, false, ipipe , false);
-           doSurfaceCheck && checkForOverlaps( pv, config, verbosityLevel>0);           
+           pv = new G4PVPlacement(rotPipe2, G4ThreeVector(0,0,z), pipe1Log, "caloPipePV", frontPanelFoamLog, false, ipipe, false);
```

Comment:
Please disambiguate all physical volume names: here caloPipePV as the ambiguity will prevent some utilities from working as they get the volumes by name

Reply:
you mean instead of caloPipePV use a unique name?

Reply:
Done